WEB/Src/PerformanceCollector/PerformanceCollector/CollectionConfiguration.cs

```diff
@@ -207,18 +231,24 @@ private void CreatePerformanceCounters(out CollectionConfigurationError[] errors
                     CollectionConfigurationError[] localErrors = null;
                     try
                     {
-                        Tuple<float, float, float, float, float> initialQuotas;
-                        previousQuotasByStreamId.TryGetValue(documentStreamInfo.Id, out initialQuotas);
+                        previousQuotasByStreamId.TryGetValue(documentStreamInfo.Id, out Tuple<float, float, float, float, float> initialQuotas);
+                        previousMaxQuotasByStreamId.TryGetValue(documentStreamInfo.Id, out Tuple<float, float, float, float, float> maxQuotas);
 
                         var documentStream = new DocumentStream(
                             documentStreamInfo,
                             out localErrors,
                             timeProvider,
-                            initialRequestQuota: initialQuotas?.Item1,
-                            initialDependencyQuota: initialQuotas?.Item2,
-                            initialExceptionQuota: initialQuotas?.Item3,
-                            initialEventQuota: initialQuotas?.Item4,
-                            initialTraceQuota: initialQuotas?.Item5);
+                            initialRequestQuota: info.InitialQuota ?? initialQuotas?.Item1,
+                            initialDependencyQuota: info.InitialQuota ?? initialQuotas?.Item2,
+                            initialExceptionQuota: info.InitialQuota ?? initialQuotas?.Item3,
+                            initialEventQuota: info.InitialQuota ?? initialQuotas?.Item4,
+                            initialTraceQuota: info.InitialQuota ?? initialQuotas?.Item5,
+                            maxRequestQuota: maxQuota ?? maxQuotas?.Item1,
+                            maxDependencyQuota: maxQuota ?? maxQuotas?.Item2,
+                            maxExceptionQuota: maxQuota ?? maxQuotas?.Item3,
+                            maxEventQuota: maxQuota ?? maxQuotas?.Item4,
+                            maxTraceQuota: maxQuota ?? maxQuotas?.Item5,
+                            quotaAccrualRatePerSec: quotaAccrualRatePerSec);
```

Comment:
What if _quotaAccrualRatePerSec_ is _null_? We have 3 values being passed in regarding quota, are all combinations valid? I suggest you validate the values somewhere at the top of this method, and make sure any error is added to the _errors_ array so that it's reported back to the user who configured it.